sympy/parsing/mathematica.py

```diff
@@ -108,6 +107,66 @@ def _deco(cls):
     return cls
 
 
+def _literal_character_ranges(initial):
+    """List all characters that may be used in identifiers
+
+    Mathematica identifiers cannot contain underscores.  This function
+    returns a regex range matching all characters that may be used in
+    a Python identifier, excluding underscores.  The `initial` argument
+    is a bool indicating whether this range describes valid initial
+    characters (True) or valid subsequent characters (False).
```

Comment:
Is it possible to use something a bit simpler for matching identifiers in the regex but then use isidentifier and check for underscores?

Reply:
I don't see how to do it.